src/main/java/com/tradeshift/flagr/Flagr.java

```diff
@@ -82,4 +83,35 @@ public EvaluationResponse evaluate(EvaluationContext context) {
         }
         return evalResponse;
     }
+
+    public Boolean evaluateBoolean(EvaluationContext context) {
+        try {
+            EvaluationResponse evalResponse = this.evaluate(context);
+            String variant = evalResponse.getVariantKey();
+            if(Boolean.parseBoolean(variant) || variant.toLowerCase().trim().equals("enabled")) {
+                return true;
+            }
+        } catch (FlagrException e) {
+            return false;
+        }
+        return false;
+    }
+
+    public String evaluateAndGetVariant(EvaluationContext context, String defaultValue) {
+        try {
+            EvaluationResponse evalResponse = this.evaluate(context);
+            return evalResponse.getVariantKey();
+        } catch (FlagrException e) {
+            return defaultValue;
+        }
+    }
+
+    public <T> T evaluateAndGetVariantAttachment(EvaluationContext context, Class<T> classOfT, T defaultValue) {
```

Comment:
Is the defaultValue really needed?

Shouldn't `evalResponse.getVariantAttachment(classOfT);` take care of that?

Reply:
@blacksails the idea is to be used like:
```java
```

Reply:
@blacksails the idea is easily get a color back without the need of an `if(color != null)`. This:
```java
Color color = flagr.evaluateAndGetVariantAttachment(
        new EvaluationContext("color"),
        Color.class,
        new Color("white", "#FFFFFF")
);
```
Instead of:
```java
Color color = flagr.evaluateAndGetVariantAttachment(
        new EvaluationContext("color"),
        Color.class
);
if (color == null) {
        color = new Color("white", "#FFFFFF");
}
```

Reply:
I find the last snippet easier to understand, because I don't need to know what the thirds argument to the method is.

Reply:
We can go with that one then.

Reply:
The second one is the goish way, whereas the first one is more javaish :wink:

Reply:
... you could also use `Optional`s for that... lots of people like that too...

Reply:
@callebjorkell I might be a little biased

Reply:
:smile: 
![image](https://user-images.githubusercontent.com/4091674/77654864-e9a34380-6f71-11ea-8007-45914bd020d9.png)

Reply:
The `Optional` approach seems cleaner to me.

Reply:
I like it! Changing again.

Reply:
I agree.. before this comment, I've create a few.. perhaps use optional here :)

Reply:
it's using Optional now